src/plugins/controls/common/options_list/types.ts

```diff
@@ -14,7 +14,10 @@ import type { DataControlInput } from '../types';
 
 export const OPTIONS_LIST_CONTROL = 'optionsListControl';
 
+export type OptionsListSearchTechnique = 'prefix' | 'wildcard';
+
 export interface OptionsListEmbeddableInput extends DataControlInput {
+  searchTechnique?: OptionsListSearchTechnique;
```

Comment:
Originally, I had this as a boolean `wildcardQuery` property that would be true only if the user selected the "contains" search technique; however, after further thought about extensibility and preventing migrations, I think it makes more sense to store this as a string value.

After all, right now we only support two search types (both of which only work for "string" type fields) - "prefix" and "contains" searching. But what about when we add support for number fields to the options list control - perhaps search techniques like "less than" and "equal to" might make more sense? Or right now, we have essentially a version of "contains" for IP fields - but maybe this doesn't always make sense, and we want to introduce other searching techniques for IP fields in the future? Having the search technique stored as a string from the very beginning rather than a boolean value allows for these changes without having to worry about a potential migration :)